### PR TITLE
skip Sentry capture for known user-state fee-estimation errors

### PR DESCRIPTION
**Pull request by B0Y3R-AVA**
## Summary

- Adds `isUserStateError(error)` to short-circuit Sentry capture in `useFeeEstimation` for expected user-state rejections: raw RPC `-32000` insufficient funds, the SDK's `causedByInsufficientFunds()` helper, and an allowlist of contract revert reasons (`ERC20: transfer amount exceeds balance/allowance`, `Png::transferFrom`, `ERC20: burn amount exceeds balance`, `TRANSFER_AMOUNT_EXCEEDS_ALLOWANCE`).
- Wraps the `causedByInsufficientFunds()` invocation in try/catch so a future SDK regression in that helper can't take down the useEffect.
- Coerces `code` via `Number()` for RPC providers that emit it as a string.
- Replaces the would-be `Logger.info` (console-only) with a Sentry breadcrumb on a new allowlisted category (`feeEstimationUserState`) so investigators retain forensic context for downstream issues without creating a Sentry issue.

## Why
Mobile already gates the swap button via `useFeeValidation → canSwap` when the user lacks native AVAX or ERC20 balance/allowance. Sentry capture of those rejection paths adds nothing — the user sees an inline validation error and adjusts. These false positives currently dominate the swap-feature Sentry issue list (~2,000 unique users / 30d across CORE-REACT-NATIVE-9X6 / A92 / A9D and the user-state subset of A8W). Investigation: https://www.notion.so/35a0e2bd518081908722e62e029170a7 (Findings revisions, Hypothesis B refuted).

The allowlist starts conservative — bug-class reverts (`TargetCallFailed`, `Panic`, `UnsupportedTokenOut`) are explicitly excluded so they continue to fire Sentry events.

## Stacked PR
Base: `worktree-fusion-presubmit-fixes-pr2` (PR #3826 — quote-staleness guard). When that merges, this PR will auto-retarget upstream.

## Notes
- 29 unit tests cover the helper (positive cases + bug-class negatives + null/primitive/throw-in-helper edge cases).
- A new `AllowedSentryBreadcrumbCategory.FeeEstimationUserState` entry was added so the breadcrumb passes the project's `beforeBreadcrumb` filter.
- `Logger.info` in this useEffect was *not* sending to Sentry by default (`Logger.info` is console-only). The breadcrumb replacement strictly *increases* forensic capture without restoring the noise.

## Dev testing

**Bitrise builds:** _Fill in after CI runs._

- [ ] iOS build: 8522
- [ ] Android build: 8523

**Steps:**
1. Reproduce the noise scenario: a wallet with 0 native AVAX (or wallet with insufficient ERC20 balance for the source token) attempts a Markr swap.
2. Confirm UI behavior is unchanged: inline validation error shown, Next button disabled.
3. After deploy, Sentry inspection: no event fires for this user-state path. The breadcrumb \"[useFeeEstimation] skipped user-state error\" appears on subsequent (non-suppressed) events from the same session.
4. Regression check: trigger a non-user-state error (e.g. a real on-chain revert with TargetCallFailed). Confirm capture STILL fires — the per-signature fingerprint from PR #3824 still works.

## Related
- Jira: [CP-14260](https://ava-labs.atlassian.net/browse/CP-14260) (3 of 3 stacked PRs)

[CP-14260]: https://ava-labs.atlassian.net/browse/CP-14260?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/core-mobile/app/new/features/swap/hooks/useFeeEstimation.helpers.ts [hunk 1]
```diff
@@ -45,3 +45,82 @@ export const getFingerprintForFeeEstimationError = (
...
+ * we haven't classified.
+ */
+export const isUserStateError = (error: unknown): boolean => {
+  if (!error || typeof error !== 'object') return false
+
+  // Raw RPC -32000 insufficient funds (the un-wrapped path that lands in 9X6).
+  // Coerce `code` via Number() because some RPC providers emit it as a string.
+  if (
+    'code' in error &&
+    Number(error.code) === -32000 &&
```
[the remark this answers is not in the document]
Comment (ruijia1in) [reply packages/core-mobile/app/new/features/swap/hooks/useFeeEstimation.helpers.ts]: I think this is valid.  can we guard Number(error.code)?

**Review by ruijia1in — APPROVED** [review]

**Review by onghwan — APPROVED** [review]
